tests/smoketests.rs

```diff
@@ -108,3 +108,17 @@ fn badssl_with_env() {
     check_site("self-signed.badssl.com").unwrap();
     env::remove_var("SSL_CERT_FILE");
 }
+
+#[test]
+#[serial]
+fn malformed_file_from_env() {
+    let site = "letsencrypt.org";
+    check_site(site).unwrap();
+
+    // Invalid data in file!() is ignored.
+    env::set_var("SSL_CERT_FILE", file!());
+    // Connect error (not cert load error).
+    assert!(check_site(site).is_err());
+
+    env::remove_var("SSL_CERT_FILE");
```

Comment:
I'll have to think about this a bit more. Yes, `load_pem_certs()` should work for most cases, I guess. Of course, this would mean the tests could no longer be integration tests (`load_pem_certs()` is a private function). This is probably alright though.

I somehow do like the idea of having something close to integration tests. Perhaps, would could do something like this:

```
pub fn load_native_certs() -> Result<Vec<CertificateDer<'static>>, Error> {
    let ssl_cert_file = env::var("SSL_CERT_FILE").map(PathBuf::from);
    let ssl_cert_dir = ...;
    load_native_certs_internal(ssl_cert_file, ssl_cert_dir);
}
```

Then we could use `load_native_certs_internal()` (better suggestions for a name welcome) in the tests.

Reply:
I have been thinking about this for a while, I believe `#[serial]` should be just fine. I realized, though, that many of the new tests just check certificate loading (e.g. ensure no certificate is found, or an error is returned). It seems rather silly, in hindsight, to use `check_site()` for those. I converted them into unit tests which turned out to be easier to read anyway.